controllers/propagator/propagation.go

```diff
@@ -901,6 +937,83 @@ func (r *PolicyReconciler) processTemplates(
 			ManagedClusterName: decision.ClusterName,
 		}
 
+		// Handle value encryption initialization
+		if encryptionEnabled {
+			usesEncryption := templates.UsesEncryption(
+				policyT.ObjectDefinition.Raw, templateCfg.StartDelim, templateCfg.StopDelim,
+			)
+			// Initialize AES Key and initialization vector
+			if usesEncryption && !encryptionInitialized {
+				log.V(1).Info("Found an object definition requiring encryption. Handling encryption keys.")
+				// Get/generate the encryption key
+				encryptionKey, err := r.getEncryptionKey(decision.ClusterName)
+				if err != nil {
+					log.Error(err, "Failed to get AES key",
+						"policy", replicatedPlc.Name, "cluster", decision.ClusterName,
+					)
+				}
+
+				// Get/generate the initialization vector
+				initializationVector, newIV, err := r.getInitializationVector(
```

Comment:
I don't think it's important that we know if the IV was generated or not. I'm thinking you can just set the annotations every time. If the IV is not new, then it's just a noop. This would make the code a little easier to understand.